### PR TITLE
Fix maven directories for publishing

### PR DESCRIPTION
**Pull request by r4stered**
This fixes the issue where new releases were going to a photonlib-* directory instead of PhotonLib-*

### CODE REVIEW

File: photon-lib/build.gradle [hunk 1]
```diff
@@ -7,7 +7,7 @@ apply plugin: "edu.wpi.first.NativeUtils"
 import java.nio.file.Path
 
 ext {
-    nativeName = "photonlib"
```
Comment (srimanachanta) [inline photon-lib/build.gradle:10]: These really should be lowercase, this would also cause the vendordep to break which we want to strickly avoid. Is there a specific reason why this change resolves the issue?

Comment (r4stered) [reply photon-lib/build.gradle:10]: [def baseArtifactId = nativeName](https://github.com/PhotonVision/photonvision/blob/68d8a943f7795a68283a714b156d043aaaa00582/shared/javacommon.gradle#L11)

This line makes the artifact id based on the nativename. Thats why this fixes the maven dir issue

Comment (mcm001) [reply photon-lib/build.gradle:10]: Why would the vendor dep break?

**Comment by mcm001** [comment]
Proposal: just change the artifact id of the json?

**Comment by mcm001** [comment]
Also keep in mind that windows is not case sensitive. If we change caps we should also change spelling

File: photon-lib/build.gradle (hunk 1 above)
Comment (srimanachanta) [reply photon-lib/build.gradle:10]: Because the vendordep is looking for that, previously it was Title case but now it is all lowercase, old vendor deps are looking for Title case, whereas all beta vendordeps are looking for lowercase. Going back would break all beta vendordeps.

**Comment by srimanachanta** [comment]
> Also keep in mind that windows is not case sensitive. If we change caps we should also change spelling

Changes between years have only been in capitalization, not spelling, windows should be unaffected?

**Comment by mcm001** [comment]
Let's change https://github.com/PhotonVision/photonvision/blob/0cec1eef9ff9ec66c1b4834e8c9fd57e345b5fae/photon-lib/build.gradle#L102 to match the old name instead.

**Comment by mcm001** [comment]
Closing for now since it seems like we just stuck with the quietly breaking change
